### PR TITLE
Render training, and minor changes for reward setup etc

### PR DESCRIPTION
**Pull request by alik-git**
moved a bunch of rendering code into helper functions in utils/visualizatios.py

I do a new unroll trajectories and save a video everytime a checkpoint is saved.

changed the xy vel tracking reward for it to work with 0 xy vel command

### CODE REVIEW

File: examples/default_humanoid/walking.py [hunk 1]
```diff
@@ -156,5 +156,8 @@ def get_init_critic_carry(self) -> None:
             num_steps_per_trajectory=600,
             minibatch_size=1024,
             num_learning_epochs=10,
+            save_every_n_steps=100,
+            save_every_n_seconds=60 * 5,
```
Comment (michael-lutz) [inline examples/default_humanoid/walking.py]: Why do we need both? Also maybe clarify in variable names what "step" means. As in minibatch steps, epochs, rollouts, datasets?

File: ksim/builders/rewards.py [hunk 2]
```diff
@@ -188,9 +188,18 @@ def __call__(
...
+        cmd_vel_xy = command_t[self.cmd_name]
+        actual_vel_xy = mjx_data_t_plus_1.qvel[:2]
+
+        # Compute tracking error as L2 distance between commanded and actual velocity
+        tracking_error = jnp.linalg.norm(cmd_vel_xy - actual_vel_xy)
+
+        # Convert error to reward in [0,1] range using exponential decay
+        # sensitivity controls how quickly reward decays with error
+        # Higher sensitivity = sharper decay = more demanding tracking
+        tracking_reward = jnp.exp(-self.sensitivity * tracking_error)
```
Comment (michael-lutz) [inline ksim/builders/rewards.py:200]: Cool idea! Have you tried this out elsewhere?

File: ksim/task/rl.py [hunk 3]
```diff
@@ -173,11 +174,59 @@ def run(self) -> None:
     # Logging and Rendering #
     #########################
 
+    def get_checkpoint_number_and_path(self) -> tuple[int, Path]:
+        """Get the checkpoint number and path from config or latest checkpoint.
+
+        Returns:
+            tuple: (checkpoint_number, checkpoint_path)
```
Comment (michael-lutz) [inline ksim/task/rl.py]: For consistency, do we want to only have one line per docstring?

File: ksim/task/rl.py [hunk 4]
```diff
@@ -173,11 +174,59 @@ def run(self) -> None:
...
+        return ckpt_num, checkpoint_path
+
     def get_render_name(self, state: xax.State | None = None) -> str:
+        """Get a unique name for the render directory based on state and checkpoint info.
+
+        Args:
+            state: Current training state if available
+
+        Returns:
+            Formatted string containing timestamp and either state step count or checkpoint number
```
Comment (michael-lutz) [inline ksim/task/rl.py]: Same here

**Review by michael-lutz — CHANGES_REQUESTED** [review]
Nice!

**Comment by alik-git** [comment]
just fyi, the test that is failing rn is for linear tracking velocity reward which I changed

File: examples/default_humanoid/walking.py (hunk 1 above)
Comment (alik-git) [reply examples/default_humanoid/walking.py]: this is what isaaaclab does for the G1 robot, see here 
https://github.com/isaac-sim/IsaacLab/blob/769d0c5d3fc0b97ee4528712ea05a2ef325ff92e/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/config/g1/rough_env_cfg.py#L24

Comment (alik-git) [reply examples/default_humanoid/walking.py]: the actual reward definition is here and it accounts for the heading of the robot 

https://github.com/isaac-sim/IsaacLab/blob/769d0c5d3fc0b97ee4528712ea05a2ef325ff92e/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/mdp/rewards.py#L86

File: ksim/builders/rewards.py (hunk 2 above)
Comment (alik-git) [reply ksim/builders/rewards.py:200]: this is what isaaaclab does for the G1 robot, see here 
https://github.com/isaac-sim/IsaacLab/blob/769d0c5d3fc0b97ee4528712ea05a2ef325ff92e/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/config/g1/rough_env_cfg.py#L24

Comment (alik-git) [reply ksim/builders/rewards.py:200]: the actual reward definition is here and it accounts for the heading of the robot 

https://github.com/isaac-sim/IsaacLab/blob/769d0c5d3fc0b97ee4528712ea05a2ef325ff92e/source/isaaclab_tasks/isaaclab_tasks/manager_based/locomotion/velocity/mdp/rewards.py#L86

Comment (alik-git) [reply ksim/builders/rewards.py:200]: simpler version of the reward is this (which is basically what I've implemented) but I will change this when I do brax rewards anyways so lets not worry about it too much

https://github.com/isaac-sim/IsaacLab/blob/769d0c5d3fc0b97ee4528712ea05a2ef325ff92e/source/isaaclab/isaaclab/envs/mdp/rewards.py#L287

File: examples/default_humanoid/walking.py (hunk 1 above)
Comment (alik-git) [reply examples/default_humanoid/walking.py]: addressed, changed it to only save every 30 mins

File: ksim/task/rl.py (hunk 3 above)
Comment (alik-git) [reply ksim/task/rl.py]: done

File: ksim/task/rl.py (hunk 4 above)
Comment (alik-git) [reply ksim/task/rl.py]: done

**Comment by alik-git** [comment]
ok @michael-lutz ready to merge imo

**Review by michael-lutz — APPROVED** [review]
